### PR TITLE
Aj ddos protection 5.0.0

### PR DESCRIPTION
**Pull request by iamjarvs**
@mehdi-juniper all aspects updated, ready for your review

### CODE REVIEW

**Review by mehdi-juniper — CHANGES_REQUESTED** [review]
Have done a first pass. I like the way the collector has been authored and described in the Readme, really nice work.
I have some questions/comments on the way probe has been designed as well as the resulted Dashboard, I left some comments as I think some adjustment might be needed there.
I created a variant of the probe/dashboard [here](https://10.28.218.3/#/blueprints/evpn-vex-virtual/analytics/dashboards/f5a7650b-db47-4d2f-ad3b-8c68664617c1) and I also uploaded it to the AI lab, hoping that the IXIA traffic gen would create some violations too. In terms of analytics, I think the probe should focus on either the `state` column (possible values are `OK` or `viol`) and/or the **Violation_Count** column, rather than on the dropped packet ones. It is possible that a siwtch has a policer where the threshold is more than one packet drop.

File: release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        "data_source": "real_time",
+        "time_series_duration": 86400,
+        "aggregation_period": 300,
+        "aggregation_type": "unset",
+        "combine_graphs": "none"
+      }
+    ],
+    [
+      {
+        "label": "DDoS Protection Violation Changes - 1Min",
```
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json]: Using your version, done

File: release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        "show_context": false,
+        "spotlight_mode": false,
+        "data_source": "real_time",
+        "time_series_duration": 86400,
+        "aggregation_period": 300,
+        "aggregation_type": "unset",
+        "combine_graphs": "none"
+      },
+      {
+        "label": "DDoS Protection Violation Changes History - 24Hrs",
```
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json]: Using your version, done

File: release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  ],
+  "predefined_dashboard": "",
+  "default": false,
+  "probes": {
+    "2a3c8c94-ff05-4e03-b9cb-2e7609ef556c": {
+      "label": "ddos-protection-protocols",
+      "description": "",
+      "processors": [
+        {
+          "name": "Extensible Service Collector",
```
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json]: Using your version, done

File: release_5.0.0/DDoS_Protection_Protocols/README.md [hunk 4]
```diff
@@ -1,3 +1,967 @@
...
+
+<br>
+
+Output stage:
+
+<img src="Images/DDoS-Protection-Protocols_Probe_Source_Processor_Output.png" width="80%" height="80%">
+
+<br>
+
+To implement the anomaly raising logic for reboot in last **1h** / **1 day** / **1 week** without duplicating the anomalies (i.e having a device which rebooted 45mn ago to appear in both 1h and 1day) we will configure the Range check processors in secitons: one section for uptime <= `60` mn, a second section for uptime > `60` mn and <= `1440` mn (total minutes count for a **day**) and a third section for uptime > `1440` mn and <= `10080` mn (total minutes count for a **week**).
```
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/README.md]: Layer 8 moment - removed

**Comment by iamjarvs** [comment]
@mehdi-juniper  please re-review, changes made

File: release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json]: Using your version, done

File: release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        "aggregation_type": "last",
+        "combine_graphs": "none"
+      }
+    ]
+  ],
+  "predefined_dashboard": "",
+  "default": false,
+  "probes": {
+    "2a3c8c94-ff05-4e03-b9cb-2e7609ef556c": {
+      "label": "ddos-protection-protocols",
```
[the remark this answers is not in the document]
Comment (iamjarvs) [reply release_5.0.0/DDoS_Protection_Protocols/Content/Final_Dashboard_Control_Plane_DDoS_Protection.json]: Using your version, done

File: release_5.0.0/DDoS_Protection_Protocols/README.md [hunk 6]
```diff
@@ -1,3 +1,1009 @@
...
+      - [Probe pipeline structure](#probe-pipeline-structure)
+      - [Probe processors details](#probe-processors-details)
+    - [Dashboards](#dashboards)
+
+<br>
+
+
+## Description of the use-case
+
+- Collect device's DDoS policer violation statistics for all protocols and raise an anomaly whenevr a protocol has an increment of the violation count.
```
Comment (mehdi-juniper) [inline release_5.0.0/DDoS_Protection_Protocols/README.md]: type o `whenever`

File: release_5.0.0/DDoS_Protection_Protocols/README.md [hunk 7]
```diff
@@ -1,3 +1,1009 @@
...
+      - [Probe processors details](#probe-processors-details)
+    - [Dashboards](#dashboards)
+
+<br>
+
+
+## Description of the use-case
+
+- Collect device's DDoS policer violation statistics for all protocols and raise an anomaly whenevr a protocol has an increment of the violation count. 
+- Keep track of these anomalies in MetricDB so user can back in time and look for any protocol DDoS violation.
```
Comment (mehdi-juniper) [inline release_5.0.0/DDoS_Protection_Protocols/README.md]: so user can `go` back ..

File: release_5.0.0/DDoS_Protection_Protocols/README.md [hunk 8]
```diff
@@ -1,3 +1,1009 @@
...
+
+### Telemetry Collectors
+
+
+> **Note**  
+> The collectors are imported as part of the service schema
+
+![DDoS-Protection-Protocols_Collector](Images/DDoS-Protection-Protocols_Collector.png)
+
+Pay attention to the expression used in the `Value` and the logic to convert the text string provided by the `policer-violation-count` field into an integer value. The  `policer-violation-count` field returns a value in a decimal format, such as `0.0`, which seems odd, considering this represent a count which in theory should inly be decimal.
```
Comment (mehdi-juniper) [inline release_5.0.0/DDoS_Protection_Protocols/README.md]: type o: should `only`

**Review by mehdi-juniper — APPROVED** [review]
Looks Great!, found few minor type-o/spelling errors, other than that it's great. Approving.

**Comment by iamjarvs** [comment]
Fixed, merged, branch removed

**Pull request merged**